receiver/opencensus/opencensus.go

```diff
@@ -53,18 +61,27 @@ var (
 	errAlreadyStopped = errors.New("already stopped")
 )
 
-const defaultOCReceiverPort = 55678
-
 // New just creates the OpenCensus receiver services. It is the caller's
 // responsibility to invoke the respective Start*Reception methods as well
 // as the various Stop*Reception methods or simply Stop to end it.
-func New(addr string) (*Receiver, error) {
+func New(grpcAddr string, httpPort int, opts ...Option) (*Receiver, error) {
 	// TODO: (@odeke-em) use options to enable address binding changes.
-	ln, err := net.Listen("tcp", addr)
+	grpcLn, err := net.Listen("tcp", grpcAddr)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to bind to gRPC address %q: error: %v", grpcAddr, err)
+	}
+
+	httpAddr := fmt.Sprintf(":%d", httpPort)
+	httpLn, err := net.Listen("tcp", httpAddr)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to bind to address %q: error: %v", addr, err)
+		return nil, fmt.Errorf("Failed to bind to HTTP port %q: error: %v", httpAddr, err)
+	}
+
+	ocr := &Receiver{grpcLn: grpcLn, httpLn: httpLn}
+
+	for _, opt := range opts {
+		opt.WithReceiver(ocr)
```

Comment:
Per discussion in other PR the `WithReceiver` should be private (this shouldn't cause problems and achieves the desired encapsulation).

Reply:
Done, makes sense.